python_modules/libraries/dagster-docker/dagster_docker/ops/docker_container_op.py

```diff
@@ -123,15 +135,21 @@ def execute_docker_container(
 
     client = _get_client(container_context)
 
-    try:
-        container = _create_container(
-            context, client, container_context, image, entrypoint, command
-        )
-    except docker.errors.ImageNotFound:
+    if pull_policy == DockerPullPolicy.ALWAYS:
         client.images.pull(image)
+
+    try:
         container = _create_container(
             context, client, container_context, image, entrypoint, command
         )
+    except docker.errors.ImageNotFound as e:
+        if pull_policy == DockerPullPolicy.IF_NOT_PRESENT:
+            client.images.pull(image)
+            container = _create_container(
+                context, client, container_context, image, entrypoint, command
+            )
+        elif pull_policy == DockerPullPolicy.NEVER:
```

Comment:
i think this should just be `else:` (it doesn't matter much in practice, but in the unusual case where you have ALWAYS set and it then fails for some reason, we would still want to raise here